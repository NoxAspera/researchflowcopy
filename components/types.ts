/**
 * Types.ts
 * @author Blake Stambaugh
 * Updated: 1/14/25 - MO
 * 
 * Helper that will contain more complicated types, like those for navigation.
 */
import { StackNavigationProp } from '@react-navigation/stack';
import { RouteProp } from '@react-navigation/native';
import { visit } from '../scripts/APIRequests';

// Define the type for the stack's navigation parameters
export type RootStackParamList = {
    // used for home screen
    SelectSite: { from: string }; 
    SelectInstrument: { from: string , notes: boolean};
    SelectTank: { from: string, onSelect?: (selectedTank: string) => void };
    SelectNotes: { from: string }; 

    // used for site select
    AddNotes: { site: string };
    AddNotesMobile: { site: string };
    ViewNotes: { site: string, from: string, visits: visit[] };
    BadData: { site: string };
    InstrumentMaintenance: { site: string };
    TankTracker: { site: string };
    PlanVisit: {site: string};
    Diagnostics: {site: string};
    ContactInfo: {site: string};

    // used for auth
    Home: undefined;
    ViewNotifications: undefined;
    NotificationsButton: undefined;
    Calendar: undefined;
};

// Type for the navigation prop for the HomeScreen
export type NaviProp = StackNavigationProp<RootStackParamList, 'SelectSite'>;

// Props for the HomeScreen component
export interface NavigationType {
  navigation: NaviProp;
}

// type for routes
type RouteParams = {
  site: string; 
  info: string; 
  from: string;
  notes: string;
  onSelect?: (selectedTank: string ) => void;
  visits: visit[];
};
export type routeProp = RouteProp<{params: RouteParams}, 'params'>;