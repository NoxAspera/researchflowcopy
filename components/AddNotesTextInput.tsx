import { View, Text, StyleSheet, ViewStyle } from "react-native";
import { TextInput } from "react-native-gesture-handler";
import { SafeAreaView, SafeAreaProvider } from "react-native-safe-area-context";

interface ANTIProp{
    inputText: string;
    inputTextVar?: string;
    inputTextStyle: ViewStyle;
    ViewStyle: ViewStyle; 
}

const AddNotesTextInput: React.FC<ANTIProp> = ({ inputText, inputTextStyle, ViewStyle, inputTextVar }) => {
    return (
        <View style = {ViewStyle}>
            <Text style = {styles.label}>{inputText}{inputTextVar}</Text>
            <SafeAreaProvider>
              <SafeAreaView>
                <TextInput
                  style = {inputTextStyle}>
                </TextInput>
              </SafeAreaView>
            </SafeAreaProvider>
          </View>
    );
};

const styles = StyleSheet.create({
    label: {
        margin: 15,
        fontSize: 24,
        alignItems: 'flex-start'
    }
  });

  export default AddNotesTextInput;