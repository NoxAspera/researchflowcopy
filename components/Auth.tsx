import { useEffect, useState } from 'react';
import * as WebBrowser from 'expo-web-browser';
import * as config from "../config"
import { makeRedirectUri, useAuthRequest, AuthSessionResult} from 'expo-auth-session';
import { generateGithubToken, readUpdates, setGithubToken, tankTrackerOffline, tankTrackerSpinUp, updateDirectories } from '../scripts/APIRequests';
import { StyleSheet, ScrollView} from 'react-native';
import { NavigationType} from './types'
import React from 'react';
import {Layout} from '@ui-kitten/components'
import HomeButtonProp from './HomeButtonProp';
import * as Network from 'expo-network'
import LoadingScreen from './LoadingScreen';


WebBrowser.maybeCompleteAuthSession();

// Endpoint
const discovery = {
  authorizationEndpoint: 'https://github.com/login/oauth/authorize',
  tokenEndpoint: 'https://github.com/login/oauth/access_token',
  revocationEndpoint: 'https://github.com/settings/connections/applications/Iv23liZ9ogrXCPdG093f',
};


async function isConnected()
{
  let check = (await Network.getNetworkStateAsync()).isConnected
  return check
}

export default function App({navigation}: NavigationType) {
  const [networkStatus, setNetworkStatus] = useState(true)
  const [loadingValue, setLoadingValue] = useState(false);

  console.log(makeRedirectUri({scheme: 'researchflowuofu'}))
  const [request, response, promptAsync] = useAuthRequest(
    {
      clientId: config.GITHUB_CLIENT_ID,
      scopes: ['repo'],
      redirectUri: makeRedirectUri({
        scheme: 'researchflowuofu'
      }),
    },
    discovery
  );

  async function handleResponse(response: AuthSessionResult | null)
  {
    if (response?.type === 'success') {
      setLoadingValue(true)
      console.log(response)
        const { code } = response.params;
        
        const {token_type, scopes, access_token} = (await generateGithubToken(code)).data
        setGithubToken(access_token);
        await tankTrackerSpinUp();
        await readUpdates()
        await updateDirectories()
        setLoadingValue(false)
    }
  }

  useEffect(() => {
    handleResponse(response)
  }, [response]);

async function handlePress()
{
  setNetworkStatus(await isConnected())
  console.log(networkStatus)
  if(networkStatus)
  {
    await promptAsync();
  }
  else
  {
    setLoadingValue(true)
    await tankTrackerOffline()
    setLoadingValue(false)
  }
  navigation.navigate("Home")
}

  return (
    <Layout style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollContainer}>
      <LoadingScreen visible={loadingValue} />
        <HomeButtonProp
          text="Login With Github"
          color= "#FFFFFF"
          onPress={() => {
            handlePress()
          }}
        />
        </ScrollView>
    </Layout>
      
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',        // has button fill space horizontally
    justifyContent: 'space-evenly',
  },
  scrollContainer: {
    paddingVertical: 16,
    alignItems: 'center', // Center cards horizontally
  },
});
