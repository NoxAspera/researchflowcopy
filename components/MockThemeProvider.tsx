import React, { useState } from "react";
import { ThemeContext } from "./ThemeContext";
import { ApplicationProvider, IconRegistry } from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { EvaIconsPack } from '@ui-kitten/eva-icons';
import customColors from '../custom-theme.json'

export const MockThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState('light');

  const currentTheme = { ...eva[theme], ...customColors };

  const toggleTheme = () => {
    setTheme((prevTheme) => (prevTheme === 'light' ? 'dark' : 'light'));
  };

  return (
    <ThemeContext.Provider value={{ theme, toggleTheme }}>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={currentTheme}>
        {children}
      </ApplicationProvider>
    </ThemeContext.Provider>
  );
};