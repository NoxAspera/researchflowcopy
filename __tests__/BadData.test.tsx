import { fireEvent, render } from '@testing-library/react-native'
import { NaviProp } from '../components/types';
import React from 'react';
import { MockThemeProvider } from '../components/MockThemeProvider';
import BadData from '../components/BadData';

beforeEach(() => {
  jest.spyOn(console, 'error')
  // @ts-ignore jest.spyOn adds this functionallity
  console.error.mockImplementation(() => null);
});

afterEach(() => {
  // @ts-ignore jest.spyOn adds this functionallity
  console.error.mockRestore()
})

const mockNavigation: NaviProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  navigateDeprecated: jest.fn(),
  preload: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getId: jest.fn(),
  getState: jest.fn(),
  setStateForNextRouteNamesChange: jest.fn(),
  getParent: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(), 
  push: jest.fn(), 
  pop: jest.fn(), 
  popToTop: jest.fn(), 
  popTo: jest.fn()
}

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
  useRoute: () => ({
    params: {},
  }),
}));

jest.mock('react-native-gesture-handler', () => {
  const { View } = require('react-native');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Others */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

describe('Bad Data', () => {
  test('renders correctly', () => {
    const { toJSON } = render(
    <MockThemeProvider>
      <BadData navigation={mockNavigation} />
    </MockThemeProvider>
  );
  expect(toJSON()).toMatchSnapshot();
  });
})