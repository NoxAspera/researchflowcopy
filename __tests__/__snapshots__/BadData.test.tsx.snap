// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bad Data renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "alignItems": "stretch",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View>
    <View
      appearance="default"
      level="1"
      style={
        [
          {
            "backgroundColor": "#FFFFFF",
          },
          {
            "alignItems": "stretch",
            "flex": 1,
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <Text
        appearance="default"
        category="h1"
        status="basic"
        style={
          [
            {
              "color": "#222B45",
              "fontFamily": "System",
              "fontSize": 36,
              "fontWeight": "800",
            },
            {
              "textAlign": "center",
            },
          ]
        }
      />
      <View
        appearance="default"
        level="1"
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "flex": 1,
            "margin": 6,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "#A0A0A0",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "800",
                  "marginBottom": 4,
                },
              ],
            ]
          }
        >
          Instrument
        </Text>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              ellipsizeMode="tail"
              numberOfLines={1}
              status="basic"
              style={
                [
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  [
                    {
                      "flex": 1,
                      "textAlign": "left",
                    },
                    {
                      "color": "#A0A0A0",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "400",
                      "marginHorizontal": 8,
                    },
                    "",
                  ],
                ]
              }
            >
              Choose an instrument
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={24}
                bbWidth={24}
                fill="#A0A0A0"
                focusable={false}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "height": 24,
                      "marginHorizontal": 8,
                      "width": 24,
                    },
                    {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4288716960,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="24"
                        opacity={0}
                        width="24"
                        x={0}
                        y={0}
                      />
                      <RNSVGPath
                        d="M12 15.5a1 1 0 0 1-.71-.29l-4-4a1 1 0 1 1 1.42-1.42L12 13.1l3.3-3.18a1 1 0 1 1 1.38 1.44l-4 3.86a1 1 0 0 1-.68.28z"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "margin": 6,
            "opacity": 1,
          }
        }
        testID="@undefined/container"
      >
        <Text
          appearance="default"
          category="p2"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Old ID
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            appearance="default"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="123456"
            placeholderTextColor="#A0A0A0"
            secureTextEntry={false}
            size="medium"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                {
                  "fontSize": 18,
                  "maxHeight": 24,
                },
              ]
            }
            testID="@undefined/input"
            value="all"
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "margin": 6,
            "opacity": 1,
          }
        }
        testID="@undefined/container"
      >
        <Text
          appearance="default"
          category="p2"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          New ID
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            appearance="default"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="67890"
            placeholderTextColor="#A0A0A0"
            secureTextEntry={false}
            size="medium"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                {
                  "fontSize": 18,
                  "maxHeight": 24,
                },
              ]
            }
            testID="@undefined/input"
            value="NA"
          />
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "margin": 6,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              [
                {
                  "color": "#A0A0A0",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "800",
                  "marginBottom": 4,
                },
                {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          Start Date
        </Text>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              ellipsizeMode="tail"
              numberOfLines={1}
              status="basic"
              style={
                [
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  {
                    "color": "#A0A0A0",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "marginHorizontal": 8,
                  },
                ]
              }
            >
              Start Date
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "margin": 6,
            "opacity": 1,
          }
        }
        testID="@undefined/container"
      >
        <Text
          appearance="default"
          category="p2"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Start Time
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            appearance="default"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="12:00:00"
            placeholderTextColor="#A0A0A0"
            secureTextEntry={false}
            size="medium"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                {
                  "fontSize": 18,
                  "maxHeight": 24,
                },
              ]
            }
            testID="@undefined/input"
            value=""
          />
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
            "margin": 6,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              [
                {
                  "color": "#A0A0A0",
                  "fontFamily": "System",
                  "fontSize": 12,
                  "fontWeight": "800",
                  "marginBottom": 4,
                },
                {
                  "textAlign": "left",
                },
              ],
            ]
          }
        >
          End Date
        </Text>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "minHeight": 40,
                "opacity": 1,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              ellipsizeMode="tail"
              numberOfLines={1}
              status="basic"
              style={
                [
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  {
                    "color": "#A0A0A0",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "normal",
                    "marginHorizontal": 8,
                  },
                ]
              }
            >
              End Date
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "margin": 6,
            "opacity": 1,
          }
        }
        testID="@undefined/container"
      >
        <Text
          appearance="default"
          category="p2"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          End Time
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            appearance="default"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="12:00:00"
            placeholderTextColor="#A0A0A0"
            secureTextEntry={false}
            size="medium"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                {
                  "fontSize": 18,
                  "maxHeight": 24,
                },
              ]
            }
            testID="@undefined/input"
            value=""
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 1,
            "margin": 6,
            "opacity": 1,
          }
        }
        testID="@undefined/container"
      >
        <Text
          appearance="default"
          category="p2"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Name
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            appearance="default"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="John Doe"
            placeholderTextColor="#A0A0A0"
            secureTextEntry={false}
            size="medium"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                {
                  "fontSize": 18,
                  "maxHeight": 24,
                },
              ]
            }
            testID="@undefined/input"
            value=""
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 4,
            "margin": 6,
            "opacity": 1,
          }
        }
        testID="@undefined/container"
      >
        <Text
          appearance="default"
          category="p2"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 13,
                "fontWeight": "400",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Reason for Bad Data
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": "#F7F9FC",
                "borderColor": "#E4E9F2",
                "borderRadius": 4,
                "borderWidth": 1,
                "minHeight": 40,
                "paddingHorizontal": 8,
                "paddingVertical": 7,
              },
              {
                "alignItems": "center",
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            appearance="default"
            editable={true}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            placeholder="Its wack."
            placeholderTextColor="#A0A0A0"
            secureTextEntry={false}
            size="medium"
            status="basic"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 15,
                  "fontWeight": "normal",
                  "marginHorizontal": 8,
                },
                {
                  "flexBasis": "auto",
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
                null,
                {
                  "fontSize": 18,
                  "maxHeight": 100,
                },
              ]
            }
            testID="@undefined/input"
            value=""
          />
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#596CA7",
            "borderColor": "#596CA7",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "margin": 15,
            "minHeight": 40,
            "minWidth": 40,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 12,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          status="basic"
          style={
            [
              {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              {
                "color": "#FFFFFF",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "bold",
                "marginHorizontal": 10,
              },
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
